imap_processing/codice/codice_l1a.py

```diff
@@ -12,17 +11,292 @@
     cdf_filename = codice_l1a(packets)
 """
 
+# TODO: Change print statements to logging
+
 import logging
+import random
+from pathlib import Path
 
 import imap_data_access
-import space_packet_parser
+import numpy as np
+import pandas as pd
+import xarray as xr
 
-from imap_processing.cdf.utils import write_cdf
-from imap_processing.codice.utils import CODICEAPID, create_dataset
+from imap_processing import imap_module_directory
+from imap_processing.cdf.global_attrs import ConstantCoordinates
+from imap_processing.codice.cdf_attrs import codice_l1a_global_attrs
+from imap_processing.codice.constants import (
+    ESA_SWEEP_TABLE_ID_LOOKUP,
+    LO_COLLAPSE_TABLE_ID_LOOKUP,
+    LO_COMPRESSION_ID_LOOKUP,
+    LO_STEPPING_TABLE_ID_LOOKUP,
+)
+from imap_processing.codice.decompress import decompress
+from imap_processing.codice.utils import (
+    CODICEAPID,
+    create_dataset,
+)
 from imap_processing.utils import group_by_apid, sort_by_time
 
 
-def codice_l1a(packets: list[space_packet_parser.parser.Packet]) -> str:
+class CoDICEL1a:
+    """Contains methods for processing L0 data and creating L1a data products.
+
+    Attributes
+    ----------
+    table_id : int
+        A unique ID assigned to a specific table configuration. This field is
+        used to link the overall acquisition and processing settings to a
+        specific table configuration
+    plan_id : int
+        The plan table that was in use.  In conjunction with ``plan_step``,
+        describes which counters are included in the data packet
+    plan_step : int
+        Plan step that was active when the data was acquired and processed. In
+        conjunction with ``plan_id``, describes which counters are included
+        in the data packet
+    view_id : int
+        Provides information about how data was collapsed and/or compressed
+    use_simulated_data : bool
+        When ``True``, simulated science data is generated and used in the
+        processing pipeline. This is useful for development and testing in the
+        absence of actual CoDICE testing data.
```

Comment:
Are you able to pass this simulated data directly in somehow instead of requiring a flag inside this class?

In the `__init__` method you are passing in only single values of ints, etc... In the real case how would you send in the bytes, or would it be processed before this step?

To give an example I'm wondering if you can do something like:
`Class(binary_data)`, then in the test suite you'd bass in `Class(_generate_simulated_data(length_in_bits))`, and in the real world you'd do something like `Class(parse_packets(packet_list))`